src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorVerifierInProcess.cs

```diff
@@ -22,6 +23,124 @@ public EditorVerifierInProcess(TestServices testServices)
+            string expectedText,
+            bool assertCaretPosition = false,
+            bool trimWhitespace = true,
+            CancellationToken cancellationToken = default)
+        {
+            if (assertCaretPosition)
+            {
+                await CurrentLineTextAndAssertCaretPositionAsync(expectedText, trimWhitespace, cancellationToken);
+            }
+            else
+            {
+                var view = await TestServices.Editor.GetActiveTextViewAsync(cancellationToken);
+                var lineText = view.Caret.Position.BufferPosition.GetContainingLine().GetText();
+
+                if (trimWhitespace)
+                {
+                    lineText = lineText.Trim();
+                }
+
+                Assert.Equal(expectedText, lineText);
+            }
+        }
+
+        private async Task CurrentLineTextAndAssertCaretPositionAsync(
+            string expectedText,
+            bool trimWhitespace,
+            CancellationToken cancellationToken)
+        {
+            var expectedCaretIndex = expectedText.IndexOf("$$");
+            if (expectedCaretIndex < 0)
+            {
+                throw new ArgumentException("Expected caret position to be specified with $$", nameof(expectedText));
+            }
+
+            var expectedCaretMarkupEndIndex = expectedCaretIndex + "$$".Length;
+
+            var expectedTextBeforeCaret = expectedText.Substring(0, expectedCaretIndex);
+            var expectedTextAfterCaret = expectedText.Substring(expectedCaretMarkupEndIndex);
+
+            var view = await TestServices.Editor.GetActiveTextViewAsync(cancellationToken);
+            var bufferPosition = view.Caret.Position.BufferPosition;
+            var line = bufferPosition.GetContainingLine();
+            var lineText = line.GetText();
+            var lineTextBeforeCaret = lineText[..(bufferPosition.Position - line.Start)];
+            var lineTextAfterCaret = lineText[(bufferPosition.Position - line.Start)..];
+
+            // Asserts below perform separate verifications of text before and after the caret.
+            // Depending on the position of the caret, if trimWhitespace, we trim beginning, end or both sides.
+            if (trimWhitespace)
+            {
+                if (expectedCaretIndex == 0)
+                {
+                    lineText = lineText.TrimEnd();
+                    lineTextAfterCaret = lineTextAfterCaret.TrimEnd();
+                }
+                else if (expectedCaretMarkupEndIndex == expectedText.Length)
+                {
+                    lineText = lineText.TrimStart();
+                    lineTextBeforeCaret = lineTextBeforeCaret.TrimStart();
+                }
+                else
+                {
+                    lineText = lineText.Trim();
+                    lineTextBeforeCaret = lineTextBeforeCaret.TrimStart();
+                    lineTextAfterCaret = lineTextAfterCaret.TrimEnd();
+                }
```

Comment:
something feels weird here, but i can't put my finger on it.  I can't determine why we would have different logic on how to trim the whitespace depending on where the caret is...  i do not like this, and it feels to me that some bug is lurking within this.

Reply:
`trimWhitespace` is a bad idea all around in these tests. I'll take this opportunity to remove it.